Add rosgraph WG charter (#11)
openrobotics/osra-policies-and-procedures: ros_project_working_group_charter_rosgraph.md

# Charter of the ROS Project Working Group "ROSGraph"

03 November 2025

### Names

The names of the Working Group shall be "ROS Working Group on ROSGraph", "WG on ROSGraph" and "ROS-WG-ROSGRAPH".

### Chair

> The Chair of the Working Group shall be [Emerson Knapp](mailto:emerson.b.knapp@gmail.com)

Reviewer: ```suggestion
The Chair of the Working Group shall be [Emerson Knapp](mailto:emerson.b.knapp@gmail.com).
```

### Purpose

> The Working Group is formed to design, plan, and build tools that help developers reason about their ROS application - which is a “Node graph” (as in graph theory, not plotting) connected by messaging interfaces- as a whole.

Reviewer: ```suggestion
The Working Group is formed to design, plan, and build tools that help developers reason about their ROS application - a “node graph” (as in graph theory, not plotting) connected by messaging interfaces - as a whole.
```

Those themes include:
- Declarative interfaces, such as generate_parameter_library and nodl
- Runtime graph and health monitoring
- Launching

> - Topic statistics

Reviewer: ```suggestion
- declarative interfaces, such as generate_parameter_library and nodl;
- runtime graph and health monitoring;
- launching; and
- topic statistics.
```

### Tasks

The Working Group shall perform the following tasks.

1. Identify usability improvements and new features needed in projects under the charter scope.
1. Build new tools where none exist to meet the determined needs.

### Expected outputs

> The Working Group shall, upon completion of its tasks and as appropriate while performing them, provide to the [Project] PMC the following.

Reviewer: ```suggestion
The Working Group shall, upon completion of its tasks and as appropriate while performing them, provide to the ROS PMC the following.
```

> 1. Bugfixes, usability improvements, features, and new tools around the WG target themes.

Reviewer: ```suggestion
1. Bug fixes, usability improvements, features, and new tools around the WG themes.
```

### Completion date

Planned for indefinite run until cancelled.

> Will discuss continuation/extension annually with the PMC, the first review of which will be scheduled for 03 November 2026.

Reviewer: ```suggestion
The Working Group's Charter expires on the 3rd of November, 2026.
```

> 

Reviewer: ```suggestion
```
